beetsplug/lyrics.py

```diff
@@ -659,26 +671,133 @@ def commands(self):
+                    )
+                if item.lyrics:
+                    if opts.printlyr:
+                        ui.print_(item.lyrics)
+                    if opts.writerst:
+                        self.writerst(opts.writerst, item)
+            if opts.writerst:
+                # flush last artist
+                self.writerst(opts.writerst, None)
+                ui.print_(u'RST files generated. to build, use one of:')
+                ui.print_(u'  sphinx-build -b html %s _build/html'
+                          % opts.writerst)
+                ui.print_(u'  sphinx-build -b epub %s _build/epub'
+                          % opts.writerst)
+                ui.print_((u'  sphinx-build -b latex %s _build/latex '
+                           u'&& make -C _build/latex all-pdf')
+                          % opts.writerst)
         cmd.func = func
         return [cmd]
 
+    def writerst(self, directory, item):
+        """Write the item to an RST file
+
+        this will keep state (in the `rst` variable) in order to avoid
+        writing continuously to the same files
+        """
+        if item is None or self.artist != item.artist:
+            if self.rst is not None:
+                slug = re.sub(r'\W+', '-', unidecode(self.artist).lower())
+                path = os.path.join(directory, 'artists', slug + u'.rst')
+                with open(path, 'wb') as output:
+                    output.write(self.rst.encode('utf-8'))
+                self.rst = None
+                if item is None:
+                    return
+            self.artist = item.artist
+            self.rst = u"%s\n%s\n\n.. contents::\n   :local:\n\n" \
+                       % (self.artist, u'=' * len(self.artist))
+        if self.album != item.album:
+            tmpalbum = self.album = item.album
+            if self.album == '':
+                tmpalbum = u'Unknown album'
+            self.rst += u"%s\n%s\n\n" % (tmpalbum,
+                                         u'-' * len(tmpalbum))
+        title_str = u":index:`%s`" % item.title
+        block = u'| ' + item.lyrics.replace(u'\n', u'\n| ')
+        self.rst += u"%s\n%s\n\n%s\n" % (title_str,
+                                         u'~' * len(title_str),
+                                         block)
+
+    def writerst_indexes(self, directory):
+        """Write conf.py and index.rst files necessary for Sphinx
+
+        We write minimal configurations that are necessary for Sphinx
+        to operate. We do not overwrite existing files so that
+        customizations are respected."""
+        try:
+            os.makedirs(os.path.join(directory, 'artists'))
+        except OSError as e:
+            if e.errno == errno.EEXIST:
+                pass
+            else:
+                raise
+        indexfile = os.path.join(directory, 'index.rst')
+        if not os.path.exists(indexfile):
+            with open(indexfile, 'wb') as output:
+                output.write(u'''Lyrics
```

Comment:
For legibility, would you mind moving these long string literals to constants at the top of the file? I find that usually makes it easier to follow the code around the strings.